build_resolvers/lib/src/resolver.dart

```diff
@@ -18,26 +19,59 @@ import 'package:path/path.dart' as native_path;
 // We should always be using url paths here since it's always Dart/pub code.
 final path = native_path.url;
 
+/// Implements [Resolver.libraries] and [Resolver.findLibraryByName] by crawling
+/// down from entrypoints.
+class PerActionResolver implements ReleasableResolver {
+  final ReleasableResolver _delegate;
+  final Iterable<AssetId> _entryPoints;
+
+  PerActionResolver(this._delegate, this._entryPoints);
+
+  @override
+  Stream<LibraryElement> get libraries async* {
+    final seen = new Set<LibraryElement>();
+    final toVisit = new Queue<LibraryElement>();
+    for (final entryPoint in _entryPoints) {
+      if (!await _delegate.isLibrary(entryPoint)) continue;
+      final library = await _delegate.libraryFor(entryPoint);
+      toVisit.add(library);
+      seen.add(library);
+    }
+    while (toVisit.isNotEmpty) {
+      final current = toVisit.removeFirst();
+      yield current;
+      final toCrawl = current.importedLibraries
+          .followedBy(current.exportedLibraries)
```

Comment:
nice, I didn't realize this had landed

Reply:
I was hesitant to use it and bump the SDK constraint, but it was too tempting to pass up :)